crates/core/tests/test_merge.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use bdk_core::Merge;
+
+#[test]
+fn test_btree_map_merge() {
+    let mut map1: BTreeMap<i32, &str> = BTreeMap::new();
+    map1.insert(1, "a");
+    let mut map2: BTreeMap<i32, &str> = BTreeMap::new();
+    map2.insert(2, "b");
+
+    map1.merge(map2);
+
+    let expected: BTreeMap<i32, &str> = BTreeMap::from([(1, "a"), (2, "b")]);
+    assert_eq!(map1, expected);
+}
+
+#[test]
+fn test_btree_set_merge() {
+    let mut set1: BTreeSet<i32> = BTreeSet::new();
+    set1.insert(1);
+    let mut set2: BTreeSet<i32> = BTreeSet::new();
+    set2.insert(2);
+
+    set1.merge(set2);
+
+    let expected: BTreeSet<i32> = BTreeSet::from([1, 2]);
+    assert_eq!(set1, expected);
+}
+
+#[test]
+fn test_vec_merge() {
+    let mut vec1 = vec![1, 2, 3];
+    let vec2 = vec![4, 5, 6];
+
+    vec1.merge(vec2);
+
+    assert_eq!(vec1, vec![1, 2, 3, 4, 5, 6]);
+}
+
+#[test]
+fn test_tuple_merge() {
+    let mut tuple1 = (vec![1, 2], BTreeSet::from([3]));
+    let tuple2 = (vec![3, 4], BTreeSet::from([4]));
+
+    tuple1.merge(tuple2);
+
+    let expected_vec = vec![1, 2, 3, 4];
+    assert_eq!(tuple1.0, expected_vec);
+    let expected_set: BTreeSet<i32> = BTreeSet::from([3, 4]);
+    assert_eq!(tuple1.1, expected_set);
+}
+
+#[test]
+fn test_is_empty() {
+    let map: BTreeMap<i32, i32> = BTreeMap::new();
+    assert!(<BTreeMap<i32, i32>>::is_empty(&map));
```

Comment:
This is testing the `BTreeMap::is_empty` method which is part of the stdlib. We want to test the `Merge::is_empty` impls on these types.

```suggestion
    assert!(Merge::is_empty(&map));
```